awscrt/s3.py

```diff
@@ -229,8 +229,6 @@ def __init__(
             region=None):
         assert isinstance(client, S3Client)
         assert isinstance(request, HttpRequest)
-        assert file is None or os.path.isfile(file)
```

Comment:
I'm not sure we want the second half of the assertion. We might be implementing our own FILE* wrappers at some point that wouldn't be valid OS paths.

Reply:
Removed! Yeah, it should not be required to be a path